vyper/parser/expr.py

```diff
@@ -382,6 +393,11 @@ def attribute(self):
             elif key == "tx.origin":
                 return LLLnode.from_list(['origin'], typ='address', pos=getpos(self.expr))
             elif key == "chain.id":
+                if get_active_evm_id() < 2:
+                    raise EvmVersionException(
+                        "chain.id is unavailable prior to istanbul ruleset",
+                        self.expr
```

Comment:
Like to have either `2` be replaced with `EVM_VERSIONS['istanbul']`, or the raw name replaced with a lookup (prefer the former)

Or even better, a helper that generates the exception based on a ruleset name

Reply:
What about a method `is_min_evm_version(evm_version)` that returns a boolean?

Reply:
It might also be cool to raise the exception here: https://github.com/vyperlang/vyper/blob/2448287461a4334b9224483f61eb3c6882dd1705/vyper/compile_lll.py#L455

It could serve as a catch-all. Then we don't need to worry about checking the validity of each specific opcode.  But I haven't checked if this might cause unintended issues elsewhere.

Reply:
I was picturing eventually it would live fully in the backend